src/Components/Components/src/NavigationManager.cs

```diff
@@ -166,6 +166,13 @@ protected virtual void NavigateToCore([StringSyntax(StringSyntaxAttribute.Uri)]
     protected virtual void NavigateToCore([StringSyntax(StringSyntaxAttribute.Uri)] string uri, NavigationOptions options) =>
         throw new NotImplementedException($"The type {GetType().FullName} does not support supplying {nameof(NavigationOptions)}. To add support, that type should override {nameof(NavigateToCore)}(string uri, {nameof(NavigationOptions)} options).");
 
+    /// <summary>
+    /// Refreshes the current page via request to the server.
+    /// If possible, statically-rendered content will be replaced without performing a full page reload.
+    /// </summary>
+    public virtual void Refresh()
+        => throw new NotImplementedException($"The type {GetType().FullName} does not implement the {nameof(Refresh)} method.");
```

Comment:
What's your opinion on having a default implementation that calls `Navigate(currentUrl, force: true)`? That would be compliant with the documented behavior, since doing an enhanced nav is best-effort.

Reply:
I like that!